src/utils/mappingUtils.js

```diff
@@ -175,14 +175,24 @@ export function autoFill(config) {
   }
 }
 
-const getCocSuggestionParams = (idx, config) => ({
+const getCocSuggestionParams = (idx, sourceDe, config) => ({
   suggestions: config.suggestions, // Because the same object has the DE and COC suggestions
-  sourceItems: config.sourceItems[idx].categoryCombo.categoryOptionCombos,
+  sourceItems: sourceDe.categoryCombo.categoryOptionCombos,
   matchThreshold: config.matchThreshold,
   setValues: config.setValues[idx].cocSetters,
   tableType: tableTypes.COC,
 })
 
+const populateCocSuggestions = (mapping, deRowIdx, notify, config) => {
+  const sourceDeUid = mapping.sourceDes[0]
+  const sourceDe = config.sourceItems.find(({ id }) => id === sourceDeUid)
```

Comment:
Is it guaranteed to find the de? If not we may want to add checks (optional chaining) in `getCocSuggestionParams` (L180). @plinnegan

Reply:
@jimiolaniyan I think this could possibly happen if the user changes the data set config or edits the data store config directly. There is no point in running this function though if the sourceDe is missing, so I'll add an error and skip the suggestion populating.